### PR TITLE
End deprecation period for using 'alias this' for partial assignment.

### PR DESCRIPTION
**Pull request by ibuclaw**
This has been deprecated in 2.086, time move it on to be a permanent error.

The comments were wrong, we still need to check the RHS for `alias this` first.  If that succeeds with a match, then that takes precedence over the partially constructed LHS `alias this` rewrite.

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

**Comment by Geod24** [comment]
You already raised a PR for this, on which I commented: https://github.com/dlang/dmd/pull/12903

**Review by Geod24 — DISMISSED** [review]
There's a false positive which would make this a regression.

**Comment by ibuclaw** [comment]
> There's a false positive which would make this a regression.

Thanks, I've already forgotten about that. The other PR is wrong anyway due to what I have mentioned in the OP.

**Comment by ibuclaw** [comment]
Need to check https://issues.dlang.org/show_bug.cgi?id=21674

**Comment by ibuclaw** [comment]
@Geod24 I had a look, and it seems to me that the conditions that check this deprecation/error are looking in the wrong place / too shallow for the intended purpose.

https://issues.dlang.org/show_bug.cgi?id=21674#c7

Would you prefer I address those comments separately, or tack them onto this PR?

**Comment by ibuclaw** [comment]
Cherry-picked regression fix -> converted to draft until the next stable->master merge.

File: changelog/alias_this_assignment.dd
Comment (ibuclaw) [reply changelog/alias_this_assignment.dd:28]: I believe the idea is that you are re-initializing it with a new value.

See also the original industry bug report: https://issues.dlang.org/show_bug.cgi?id=19441

**Review by RazvanN7 — APPROVED** [review]
Anything else blocking this?

**Comment by ibuclaw** [comment]
@RazvanN7 Waiting for merge stable->master to go in, then I'll rebase.

**Pull request merged**
